### ISSUE TITLE
train/val/test splits

### ISSUE THREAD

**Issue by MartinHahner** [mpitropov/cadc_devkit#15]
Hi Matthew,

first of all, thank you for sharing the valuable CADC data with the research community!

I have some questions regarding splitting the dataset into different subsets:

- Why did you choose specifically [these](https://github.com/mpitropov/OpenPCDet/blob/7fc4ed7f229a80ad3afdad8d6c00407cd6e8edc2/data/cadc/ImageSets/generate_splits.py#L5-L18) sequences as validation split?
Are they recorded in a different geographic location compared to the remaining sequences in the training split 
or what is the reasoning behind choosing exactly those sequences?

- Besides having a training and validation split, I also would like to have a test split.
Which sequences would you suggest taking away from the validation (and/or training) set to form a test set? 

- Finally, I am curious if there is a way to find the frames belonging to the subsets given in the figure below 
without running DROR on the entire dataset and bin them according to your explanation in the paper.
Unfortunately, I could neither find those "labels" in the _labeled_ nor the _raw_ data release.
![image](https://user-images.githubusercontent.com/14181188/120598638-c82bca80-c446-11eb-9ef4-99e3773030bf.png)

Looking forward to your reply. :slightly_smiling_face: 

Greets, Martin

**Comment by MartinHahner** [comment]
>    Finally, I am curious if there is a way to find the frames belonging to the subsets given in the figure below
>       without running DROR on the entire dataset and bin them according to your explanation in the paper.
>       Unfortunately, I could neither find those "labels" in the _labeled_ nor the _raw_ data release.
>       ![image](https://user-images.githubusercontent.com/14181188/120598638-c82bca80-c446-11eb-9ef4-99e3773030bf.png)

OK, I just found https://github.com/mpitropov/cadc_devkit/issues/6#issuecomment-639831317 to answer my last question.
Still curious about the other questions though.

**Comment by mpitropov** [comment]
I chose the split to have about an equal number of different classes of objects and snowfall levels in the different splits. Both the training and validation splits have frames from all the days I recorded.

To create a test set I think you could split the validation set into two:
```
val_set = [
  ['2018_03_06', '0001'],
  ['2018_03_06', '0016'],
  ['2019_02_27', '0009'],
  ['2019_02_27', '0028'],
  ['2019_02_27', '0040'],
  ['2019_02_27', '0054'],
  ['2019_02_27', '0065'],
]
test_val = [
  ['2018_03_06', '0008'],
  ['2018_03_07', '0004'],
  ['2019_02_27', '0016'],
  ['2019_02_27', '0033'],
  ['2019_02_27', '0043'],
  ['2019_02_27', '0060'],
  ['2019_02_27', '0076'],
]
```

**Comment by MartinHahner** [comment]
> I chose the split to have about an equal number of different classes of objects and snowfall levels in the different splits. Both the training and validation splits have frames from all the days I recorded.

Wouldn't it be better to geographically separate them?

Or separate them based on the day of recording?
- val: 2018_03_06 (12 sequences)
- test: 2018_03_07 (6 sequences)
- train: 2019_02_27 (57 sequences)

**Comment by mpitropov** [comment]
03_06 and 03_07 have much more rain/snow covering on the front camera lens compared to 02_27.

All sequences were collected in Waterloo and there is some overlap geographically but most sequences are in different locations so I chose to focus on the objects within the sequences.
